move/walrus_site/tests/site_tests.move

```diff
@@ -0,0 +1,79 @@
+#[test_only]
+module walrus_site::site_tests {
+    use walrus_site::site::{
+        ERangeEndNotPositive,
+        ERangeStartGreaterThanRangeEnd
+    };
+    #[test]
+    fun test_new_resource_no_range_bounds_defined() {
+        walrus_site::site::new_resource(
+            b"/index.html".to_string(),
+            12121212,
+            89898989,
+            option::none(),
+            option::none()
+        );
+    }
+    #[test]
+    fun test_new_resource_both_range_bounds_defined() {
+        walrus_site::site::new_resource(
+            b"/index.html".to_string(),
+            12121212,
+            89898989,
+            option::some(0),
+            option::some(1)
+        );
+    }
+    #[test]
+    fun test_new_resource_range_only_upper_bound_defined() {
+        walrus_site::site::new_resource(
+            b"/index.html".to_string(),
+            12121212,
+            89898989,
+            option::none(),
+            option::some(1024)
+        );
+    }
+    #[test]
+    fun test_new_resource_range_only_lower_bound_defined() {
+        walrus_site::site::new_resource(
+            b"/index.html".to_string(),
+            12121212,
+            89898989,
+            option::some(1024),
+            option::none()
+        );
+    }
+    #[test]
+    fun test_new_resource_range_lower_bound_can_be_zero() {
+        walrus_site::site::new_resource(
+            b"/index.html".to_string(),
+            12121212,
+            89898989,
+            option::some(0),
+            option::none()
+        );
+    }
+    #[test]
+    #[expected_failure(abort_code = ERangeEndNotPositive)]
+    fun test_new_resource_range_upper_bound_cannot_be_zero() {
+        walrus_site::site::new_resource(
+            b"/index.html".to_string(),
+            12121212,
+            89898989,
+            option::none(),
+            option::some(0)
```

Comment:
Remark: this is a bit weird: `some(0), none` (full range)  is allowed, but `none, some(0)` (empty range)

Reply:
`none, some(0)` is not allowed, as it would imply an empty range, correct. But what is the problem?

Reply:
After our call: I will enable support for empty files, since there are cases where there are empty files in folders!